tests/integration/records/table_timezone_validator.py

```diff
@@ -115,12 +115,19 @@ def validate(self,
               self.tc.variant_doesnt_support_seconds(self.file_variant)))):
             seconds = '00'
             micros = '000000'
+
+            # Some databases on rendering this as a string will just
+            # drop the microseconds if they're all zeros.
```

Comment:
As noted in the logic above this, some CSV variants don't support milli/micro-seconds, and it looks like the increased testing here triggered exactly such a path with Redshift (yay, more testing!).